src/BrazeBannerComponent.ts

```diff
@@ -1,32 +1,39 @@
 import React from 'react';
-import {
-    COMPONENT_NAME as DIGITAL_SUBSCRIBER_APP_BANNER_NAME,
-    DigitalSubscriberAppBanner,
-} from './DigitalSubscriberAppBanner';
-import { COMPONENT_NAME as APP_BANNER_NAME, AppBanner } from './AppBanner';
-import { COMPONENT_NAME as BANNER_WITH_LINK_NAME, BannerWithLink } from './BannerWithLink';
-import type { BrazeClickHandler, SubmitComponentEvent } from './utils/tracking';
+
+// Common newsletter imports
+import { NewsletterSubscribeCallback } from './newsletterCommon/sharedComponents';
+
+// Common Banner imports
 import {
     buildBrazeMessageComponent,
     ComponentMapping,
     HasConsolidatedTrackClick,
+    BrazeMessageProps,
 } from './buildBrazeMessageComponent';
+import type { BrazeClickHandler, SubmitComponentEvent } from './utils/tracking';
 
-type BrazeMessageProps = {
-    [key: string]: string | undefined;
-};
+// Current Banner components
+import {
+    COMPONENT_NAME as DIGITAL_SUBSCRIBER_APP_BANNER_NAME,
+    DigitalSubscriberAppBanner,
+} from './DigitalSubscriberAppBanner';
+import { COMPONENT_NAME as APP_BANNER_NAME, AppBanner } from './AppBanner';
+import { COMPONENT_NAME as BANNER_WITH_LINK_NAME, BannerWithLink } from './BannerWithLink';
+import { COMPONENT_NAME as BANNER_NEWSLETTER_NAME, BannerNewsletter } from './BannerNewsletter';
 
 export type CommonBannerComponentProps = {
     componentName: string;
+    brazeMessageProps: BrazeMessageProps;
+    subscribeToNewsletter: NewsletterSubscribeCallback;
```

Comment:
Just a reminder (although TypeScript will let you know too!) that this will need to be added as a prop provided by the platform on DCR and frontend, like we do for newsletters [here](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/components/SlotBodyEnd/BrazeEpic.tsx#L130).

Reply:
I think I'd do that work in a single PR on each of the web platforms. So once the newsletter banner changes are in a release, create a PR on each of frontend and DCR which 1) bumps the version of braze-components and 2) starts threading through the `subscribeToNewsletter` prop for banners. I can help with integration testing etc in CODE when we're ready to do that!